### PR TITLE
Fix closure compiler warnings with WASM_BIGINT. NFC

### PR DESCRIPTION
**Pull request by sbc100**
See #12728

### CODE REVIEW

File: src/library_wasi.js [hunk 1]
```diff
@@ -138,6 +138,7 @@ var WasiLibrary = {
   clock_time_get__nothrow: true,
   clock_time_get__sig: 'iiiii',
   clock_time_get__deps: ['emscripten_get_now', 'emscripten_get_now_is_monotonic', '$checkWasiClock'],
+  clock_time_get__docs: {{{ documentI64Param('precision') }}},
   clock_time_get: function(clk_id, {{{ defineI64Param('precision') }}}, ptime) {
```
Comment (kripken) [inline src/library_wasi.js:142]: Could the docs be inline inside `defineI64Param`? I'm hoping closure will parse a comment right before the parameter and not just a comment before the entire function...

File: src/parseTools.js [hunk 2]
```diff
@@ -1052,13 +1052,20 @@ function hasExportedFunction(func) {
   return WASM_EXPORTS.has(_asmjsDemangle(func));
 }
 
+function documentI64Param(name) {
+  if (WASM_BIGINT) {
+    return `'/** @param {!BigInt} ${name}_bigint */'`;
```
Comment (kripken) [inline src/parseTools.js:1057]: what error does this prevent?

File: src/library_wasi.js (hunk 1 above)
Comment (sbc100) [reply src/library_wasi.js:142]: I tried that for quite a long time .. and I couldn't make it work at all.   I caused the library file to fail to parse.  I will try one more time.

File: src/parseTools.js (hunk 2 above)
Comment (sbc100) [reply src/parseTools.js]: Lots of this stuff:

```
/tmp/emscripten_temp_337t15lk/hello_world.js:14230:30: WARNING - [JSC_BAD_TYPES_FOR_BINARY_OPERATION] operator & cannot be applied to ? and bigint
  14230|       var offset_low = Number(offset_bigint & BigInt(0xffffffff)) | 0, offset_high = Number(offset_bigint >> BigInt(32)) | 0;
                                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Unless closure knows that `offset_bigint` is for sure a BigInt it won't let you `&` with a `BigInt`

Comment (kripken) [reply src/parseTools.js]: Huh... closure is pretty strict then, more than I'd expect...

File: src/parseTools.js [hunk 3]
```diff
@@ -1056,9 +1056,9 @@ function hasExportedFunction(func) {
 // it is a BigInt. Otherwise, we legalize into pairs of i32s.
 function defineI64Param(name) {
   if (WASM_BIGINT) {
-    return name + '_bigint';
+    return `/** @type {!BigInt} */ ${name}_bigint`;
```
Comment (kripken) [inline src/parseTools.js:1059]: nice that this works!

File: src/library_syscall.js [hunk 4]
```diff
@@ -1201,13 +1201,10 @@ function wrapSyscallFunction(x, library, isWasi) {
   }
   post = handler + post;
 
-  if (pre) {
-    var bodyStart = t.indexOf('{') + 1;
-    t = t.substring(0, bodyStart) + pre + t.substring(bodyStart);
-  }
-  if (post) {
-    var bodyEnd = t.lastIndexOf('}');
-    t = t.substring(0, bodyEnd) + post + t.substring(bodyEnd);
```
Comment (kripken) [inline src/library_syscall.js:1210]: unrelated trivial refactor? seems trivial so sgtm to leave it in, just checking.

**Review by kripken — APPROVED** [review]

File: src/library_syscall.js (hunk 4 above)
Comment (sbc100) [reply src/library_syscall.js:1210]: No.. the old code breaks because we have a `{` which is in a comment.. but `modifyFunction` can handle it... took me a while to find that one!

Comment (kripken) [reply src/library_syscall.js:1210]: Oh cool, good find + fix!
